src/shop-app.html

```diff
@@ -18,11 +18,25 @@
 <link rel="import" href="../bower_components/iron-media-query/iron-media-query.html">
 <link rel="import" href="../bower_components/iron-pages/iron-pages.html">
 <link rel="import" href="../bower_components/iron-selector/iron-selector.html">
+<link rel="import" href="../bower_components/paper-icon-button/paper-icon-button.html">
 <link rel="import" href="shop-category-data.html">
 <!-- Since 'home' is the default route, eagerly load it. -->
 <link rel="import" href="shop-home.html">
 
+
+
 <dom-module id="shop-app">
+  <!--
+    TODO(ricitc): make these lazy-imports after
+        https://github.com/Polymer/polymer-analyzer/pull/543 lands
+        DO NOT MERGE until then :)
+  -->
+  <link rel="import" href="shop-cart-data.html">
+  <link rel="import" href="lazy-resources.html">
+  <link rel="import" href="shop-list.html">
+  <link rel="import" href="shop-detail.html">
+  <link rel="import" href="shop-cart.html">
+  <link rel="import" href="shop-checkout.html">
```

Comment:
By fragments do you mean the DOM to display for each possible choice in iron-pages?

IMO the routes, the per-page DOM, and the urls are three separate concerns, so I'm not concerned that all three need to be specified, but I am a little uncomfortable with how they're specified in such different places such that it's easy for them to get out of sync.

IMO your [`iron-lazy-pages`](https://github.com/TimvdLippe/iron-lazy-pages) element goes in the right direction, by specifying the per-page dom in the same place that you specify the url that needs to be imported to render that page.